neurolib/models/multimodel/builder/thalamus.py

```diff
@@ -0,0 +1,442 @@
+        return 20.0 + 1000.0 / (exp((voltage + 71.5) / 14.2) + exp(-(voltage + 89.0) / 11.6))
+
+    def _P_h(self, ca_conc):
+        return (self.params["k1"] * ca_conc ** self.params["n_P"]) / (
+            self.params["k1"] * ca_conc ** self.params["n_P"] + self.params["k2"]
+        )
+
+    def _derivatives(self, coupling_variables):
+        (
+            voltage,
+            ca_conc,
+            h_T,
+            m_h1,
+            m_h2,
+            syn_ext,
+            syn_inh,
+            dsyn_ext,
+            dsyn_inh,
+            firing_rate,
+        ) = self._unwrap_state_vector()
+        # voltage dynamics
+        d_voltage = -(
+            self._get_leak_current(voltage)
+            + self._get_excitatory_current(voltage, syn_ext)
+            + self._get_inhibitory_current(voltage, syn_inh)
+            + self.params["ext_current"]
+        ) / self.params["tau"] - (1.0 / self.params["C_m"]) * (
+            self._get_potassium_leak_current(voltage)
+            + self._get_T_type_current(voltage, h_T)
+            + self._get_anomalous_rectifier_current(voltage, m_h1, m_h2)
+        )
+        # calcium concetration dynamics
+        d_ca_conc = (
+            self.params["alpha_Ca"] * self._get_T_type_current(voltage, h_T)
+            - (ca_conc - self.params["Ca_0"]) / self.params["tau_Ca"]
+        )
+
+        # channel dynamics: T-type and rectifier current
+        d_h_T = (self._h_inf_T(voltage) - h_T) / self._tau_h_T(voltage)
+        d_m_h1 = (
+            (self._m_inf_h(voltage) * (1.0 - m_h2) - m_h1) / self._tau_m_h(voltage)
+            - self.params["k3"] * self._P_h(ca_conc) * m_h1
+            + self.params["k4"] * m_h2
+        )
+        d_m_h2 = self.params["k3"] * self._P_h(ca_conc) * m_h1 - self.params["k4"] * m_h2
+
+        # synaptic dynamics
+        d_syn_ext = dsyn_ext
+        d_syn_inh = dsyn_inh
+        d_dsyn_ext = (
+            self.params["gamma_e"] ** 2
+            * (
+                coupling_variables["node_exc_exc"]
+                + coupling_variables["network_exc_exc"]
+                + system_input(self.noise_input_idx[0])
+                - syn_ext
+            )
+            - 2 * self.params["gamma_e"] * dsyn_ext
+        )
+        d_dsyn_inh = (
+            self.params["gamma_r"] ** 2 * (coupling_variables["node_exc_inh"] - syn_inh)
+            - 2 * self.params["gamma_r"] * dsyn_inh
+        )
+        # firing rate as dummy dynamical variable with infinitely fast
+        # fixed-point dynamics
+        firing_rate_now = self._get_firing_rate(voltage)
+        d_firing_rate = -self.params["lambda"] * (firing_rate - firing_rate_now)
+
+        return [
+            d_voltage,
+            d_ca_conc,
+            d_h_T,
+            d_m_h1,
+            d_m_h2,
+            d_syn_ext,
+            d_syn_inh,
+            d_dsyn_ext,
+            d_dsyn_inh,
+            d_firing_rate,
+        ]
+
+
+class ThalamicReticularPopulation(ThalamicMass):
```

Comment:
```suggestion
class ThalamicReticularMass(ThalamicMass):
```
see comment upstairs :)